R/vpcstats.R

```diff
@@ -723,8 +723,8 @@ predcorrect.tidyvpcobj <- function(o, pred, data=o$data, ..., log=FALSE) {
     o$obs[, ypc := (mpred - pred) + y]
     o$sim[, ypc := (mpred - pred) + y]
   } else {
-    o$obs[, ypc := (mpred/pred)*y]
-    o$sim[, ypc := (mpred/pred)*y]
+    o$obs[, ypc := ifelse(pred == 0, 0, (mpred/pred)*y)]
+    o$sim[, ypc := ifelse(pred == 0, 0, (mpred/pred)*y)]
```

Comment:
Thanks @billdenney for the PR! Yes, we will certainly want to prevent division by zero here, however, I receive the following error in this line of code.
```
Error in `[.data.table`(o$sim, , `:=`(ypc, ifelse(pred == 0, 0, (mpred/pred) *  : 
  Supplied 600 items to be assigned to 18000 items of column 'ypc'. If you wish to 'recycle' the RHS please use rep() to make this intent clear to readers of your code.
```

It looks like we may want to update this line to:

```
o$sim[, ypc := ifelse(rep(pred, times = nrow(o$sim)/nrow(o$obs)) == 0, 0, (mpred/pred)*y)]
```

Reply:
Thanks for finding that.  I think that I was using too small a toy example in my test.